### PR TITLE
server: Fix exception while update domain resource count

### PR DESCRIPTION
**Pull request by ustcweizhou**
While update resource count of a domain, I got the following exception which should not happen

2019-03-05 08:23:43,656 DEBUG [c.c.u.d.T.Transaction] (qtp858242339-15:ctx-13652208 ctx-56fe3704) (logid:130b0537) Rolling back the transaction: Time = 10 Name =  qtp858242339-15; called by -TransactionLegacy.rollback:890-TransactionLegacy.removeUpTo:833-TransactionLegacy.close:657-TransactionContextInterceptor.invoke:36-ReflectiveMethodInvocation.proceed:174-ExposeInvocationInterceptor.invoke:92-ReflectiveMethodInvocation.proceed:185-JdkDynamicAopProxy.invoke:212-$Proxy42.countCpuNumberAllocatedToAccount:-1-ResourceLimitManagerImpl.countCpusForAccount:950-ResourceLimitManagerImpl.recalculateAccountResourceCount:910-ResourceLimitManagerImpl$3.doInTransaction:872

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by GabrielBrascher** [comment]
@blueorangutan package

File: engine/schema/src/com/cloud/configuration/dao/ResourceCountDaoImpl.java [hunk 1]
```diff
@@ -276,17 +276,18 @@ public long countMemoryAllocatedToAccount(long accountId) {
...
             pstmt.setLong(1, accountId);
 
             ResultSet rs = pstmt.executeQuery();
             if (!rs.next()) {
-              throw new CloudRuntimeException(String.format("An unexpected case happened while counting allocated computing resources for account: " + accountId));
+                return 0L;
             }
             return rs.getLong("total");
         } catch (SQLException e) {
             throw new CloudRuntimeException(e);
```
Comment (GabrielBrascher) [inline engine/schema/src/com/cloud/configuration/dao/ResourceCountDaoImpl.java]: What about adding a message similar to the "An unexpected case happened while counting allocated..." to the exception stack?

Comment (ustcweizhou) [reply engine/schema/src/com/cloud/configuration/dao/ResourceCountDaoImpl.java]: @GabrielBrascher actually it is not an error. rs.next() is false means the sql query returns null (for example, no vm in the domain/account). If there are something wrong in the sql query, they are handled in line 290.

Comment (GabrielBrascher) [reply engine/schema/src/com/cloud/configuration/dao/ResourceCountDaoImpl.java]: @ustcweizhou I was referring to the exception "CloudRuntimeException(e)" at line 290. Considering an exception "user-friendly". But it is a minor detail, overall it LGTM.

Ex: `throw new CloudRuntimeException(String.format("Failed to count allocated computing resources for account [id:%s]", accountId), e);`.

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
reviewed the test error, it is a failing ssh connection, nothing to do with this

**Pull request merged**